packages/jellyfish-testing/src/icxorderbook.ts

```diff
@@ -0,0 +1,209 @@
+import { Testing } from './index'
+import BigNumber from 'bignumber.js'
+import { icxorderbook } from '@defichain/jellyfish-api-core'
+
+export class TestingICX {
+  public symbolDFI: string
+  public symbolBTC: string
+  public accountDFI: string
+  public accountBTC: string
+  public poolOwner: string
+  public idDFI: string
+  public idBTC: string
+  public ICX_TAKERFEE_PER_BTC: BigNumber
+  public DEX_DFI_PER_BTC_RATE: BigNumber
+
+  constructor (
+    private readonly testing: Testing
+  ) {
+    this.accountDFI = ''
+    this.accountBTC = ''
+    this.poolOwner = ''
+    this.idDFI = ''
+    this.idBTC = ''
+    this.ICX_TAKERFEE_PER_BTC = new BigNumber(0)
+    this.DEX_DFI_PER_BTC_RATE = new BigNumber(0)
+    this.symbolDFI = 'DFI'
+    this.symbolBTC = 'BTC'
+  }
+
+  async setup (): Promise<void> {
+    await this.createAccounts()
+    await this.testing.rpc.account.utxosToAccount({ [this.accountDFI]: `${500}@${this.symbolDFI}` })
+    await this.testing.rpc.account.utxosToAccount({ [this.accountBTC]: `${10}@${this.symbolDFI}` }) // for fee
+    await this.testing.container.generate(1)
+
+    await this.testing.fixture.createPoolPair({
+      a: { amount: '1', symbol: this.symbolBTC },
+      b: { amount: '100', symbol: this.symbolDFI }
+    })
+    this.DEX_DFI_PER_BTC_RATE = new BigNumber(100)
+
+    await this.initializeTokensIds()
+    await this.setTakerFee(new BigNumber(0.001))
+    await this.setBTCDFIPoolOwner()
```

Comment:
can remove this also. the idea is that pool creation is separate from ICX, but the `DEX_DFI_PER_BTC_RATE ` which is set based on the pool is required for ICX inside defid.

Reply:
It seemed to me that the setup point is to set all public field, isn't it ?

Reply:
most of them yes. but here if you check, pool owner is actually not used in the tests.